libraries/AP_BoardConfig/px4_drivers.cpp

```diff
@@ -766,6 +765,87 @@ void AP_BoardConfig::px4_setup_peripherals(void)
 }
 
 /*
+  check a SPI device for a register value
+ */
+bool AP_BoardConfig::spi_check_register(const char *devname, uint8_t regnum, uint8_t value, uint8_t read_flag)
+{
+    auto dev = hal.spi->get_device(devname);
+    if (!dev) {
+        printf("%s: no device\n", devname);
+        return false;
+    }
+    dev->set_read_flag(read_flag);
+    uint8_t v;
+    if (!dev->read_registers(regnum, &v, 1)) {
+        printf("%s: reg %02x read fail\n", devname, (unsigned)regnum);
+        return false;
+    }
+    printf("%s: reg %02x %02x %02x\n", devname, (unsigned)regnum, (unsigned)value, (unsigned)v);
+    return v == value;
+}
+
+#define MPUREG_WHOAMI 0x75
+#define MPU_WHOAMI_MPU60X0  0x68
+#define MPU_WHOAMI_MPU9250  0x71
+#define MPU_WHOAMI_ICM20608 0xaf
+
+#define LSMREG_WHOAMI 0x0f
+#define LSM_WHOAMI_LSM303D 0x49
+
+/*
+  auto-detect board type
+ */
+void AP_BoardConfig::px4_autodetect(void)
+{
+    if (px4.board_type == PX4_BOARD_TEST_V1 ||
+        px4.board_type == PX4_BOARD_TEST_V2 ||
+        px4.board_type == PX4_BOARD_TEST_V3) {
+        // these were old test values
+        px4.board_type.set(0);
```

Comment:
I'll probably remove them in the future, its really just for the convenience of the few people testing now